ambry-api/src/main/java/com.github.ambry/protocol/RequestAPI.java

```diff
@@ -21,6 +21,14 @@
  * This defines the server request API. The commands below are the requests that can be issued against the server
  */
 public interface RequestAPI {
+
+  /**
+   * Handle a request.
+   * @param request The request to handle.
+   * @throws InterruptedException
+   */
+  void handleRequests(Request request) throws InterruptedException;
+
   /**
    * Puts a blob into the store. It accepts a blob property, user metadata and the blob
    * as a stream and stores them
```

Comment:
If you want, you could actually use default methods in this interface to throw the UnsupportedOperationExceptions.

For example: 
```
default void handlePutRequest(Request) throws ... {
  throw new UnsupportedOperationException("Put request not supported");
}
```
You could also consider adding handleAdminRequest to this interface. Then AmbryServerRequests would not have to override the handleRequests of AmbryRequests.

Reply:
Added admin handler as default method.  Left others the same.